client/src/routes/PageSearch.js

```diff
@@ -18,6 +18,36 @@ import SearchEasterEggs from '../utilities/SearchEasterEggs';
 
 
 class PageSearch extends React.Component {
+  async componentDidMount() {
+    const p = this.props;
+
+    if(window.location.search.length !== 0) {
+      const urlParams = new URLSearchParams(window.location.search);
+
+      if (!p.search.filterOptions) {
+        await p.dispatch(fetchFilterOptions())
+      }
+
+      for(const [key, value] of urlParams.entries()) {
+        if(key !== 'query') {
+          if(key === "category" || key === "subcategory" || key === "tournament") {
+            p.dispatch(updateSearchFilter(key, value.split(',').map(e => parseInt(e, 10))))
+          } else {
+            p.dispatch(updateSearchFilter(key, value.split(',')))
+          }
+        } else {
+          p.dispatch(updateSearch(value))
+        }
+      }
```

Comment:
For code clarity, I'd extract this block into a function that takes the search string and carries out the search filter updates. That way it can easily be extracted if in the future we want to do similar url search query -> filter updates. (Which we probably will, since the search filter will exist on other pages!)

Reply:
Would you extract to actions.js?

Reply:
Up to you--I might leave it in this component for now, since it's pretty local, and we can always extract further in the future.